src/App.js

```diff
@@ -1,19 +1,145 @@
 import React from 'react';
 import './App.css';
+import { TodoList } from './Components/TodoList';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    user: '',
+    title: '',
+    isTitle: false,
+    isUserName: false,
+  }
+
+  addNewTodo = () => {
+    this.setState(state => ({
+      todos: [...state.todos, {
+        id: Math.max(...state.todos.map(item => item.id)) + 1,
+        title: state.title,
+        completed: false,
+        user: users.find(user => user.name === state.user),
+      }],
+    }));
+  }
+
+  handlerEvent = (event) => {
+    const { user, title } = this.state;
+
+    event.preventDefault();
+
+    this.setState({
+      isUserName: !user,
+    });
+
+    if (title.length === 0) {
+      this.setState({
+        isTitle: true,
+      });
+
+      return;
+    }
```

Comment:
this.setState(state => ({
  isUserName: !state.user,
  isTitle: !state.title,
}));

if (!title) {
  return;
}

You can write it like this, but your idea is fine too)